railgun/test/integration/perf_ingress_test.go

```diff
@@ -0,0 +1,129 @@
+//+build integration_tests
+
+package integration
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kong/kubernetes-ingress-controller/pkg/annotations"
+	k8sgen "github.com/kong/kubernetes-testing-framework/pkg/generators/k8s"
+)
+
+func TestIngressPerf(t *testing.T) {
+	t.Log("setting up the TestIngressPerf")
+	proxyReady()
+
+	ctx := context.Background()
+
+	cnt := 1
+	cost := 0
+	for cnt < 1 {
+
+		namespace := fmt.Sprintf("ingress-%d", cnt)
+		nsName := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespace,
+			},
+		}
+		_, err := cluster.Client().CoreV1().Namespaces().Create(context.Background(), nsName, metav1.CreateOptions{})
+		assert.NoError(t, err)
+
+		t.Logf("[%s] deploying a minimal HTTP container deployment to test Ingress routes", namespace)
+		err = configManifest(ctx, namespace, "httpbin.yaml", t)
+		assert.NoError(t, err)
+
+		t.Logf("[%s] List the service from the namespace.", namespace)
+		svcs, err := cluster.Client().CoreV1().Services(namespace).List(ctx, metav1.ListOptions{})
+		if err != nil {
+			t.Logf("Failed to list services.")
+			continue
+		}
+
+		var service *v1.Service
+		for _, svc := range svcs.Items {
+			fmt.Printf("\n service %v \n", svc)
+			if svc.Name == "httpbin" {
+
+				service = &svc
+				break
+			}
+		}
+
+		t.Logf("[%s] creating an ingress for service httpbin with ingress.class %s", namespace, ingressClass)
+		ingress := k8sgen.NewIngressForService("/httpbin", map[string]string{
+			annotations.IngressClassKey: ingressClass,
+			"konghq.com/strip-path":     "true",
+		}, service)
+		ingress, err = cluster.Client().NetworkingV1().Ingresses(namespace).Create(ctx, ingress, metav1.CreateOptions{})
+		assert.NoError(t, err)
+		start_time := time.Now().Nanosecond()
+		end_time := start_time
```

Comment:
Rather than initialize the `end_time` here (which could produce side effects later on if the tests manage to never populate it) we should provide the normal "zero value":
```suggestion
		var end_time *time.Time
```
This will give us the opportunity in later code to validate that it becomes populated.